### ISSUE TITLE
Incorrect behavior of RemoveOverriddenValuesRector

### ISSUE THREAD

**Issue by brucealdridge** [rectorphp/rector#5033]
# Bug Report

| Subject        | Details              |
| :------------- | :--------------------|
| Rector version | last dev-master      |
| Installed as   | composer dependency  |

## Minimal PHP Code Causing Issue

See https://getrector.org/demo/f09f2811-8586-4fd3-a427-ec093ef87ad4

```php
<?php

final class DemoFile
{
    public function run()
    {
         $url = 'https://example.com/';
         $api_key = MY_API_KEY ?? null;
         //$term = $term;
         $url = $url.'?'.http_build_query(['key' => $api_key]);
    }
}
```

### Responsible rules

* `RemoveOverriddenValuesRector`

## Expected Behavior

initial definition of $url is removed. Seems like the comment breaks this behaviour. Remove the comment and all is good, so maybe the comment is being parsed incorrectly?

**Comment by samsonasik** [comment]
If commented code: 

```php
         //$term = $term;
```

The result is valid https://getrector.org/demo/95cec5af-6513-4fa4-84c1-b6b857c068ef . It may related with comment handling somewhere.

**Comment by samsonasik** [comment]
Fixed by #5105
